src/layer_norm.cpp

```diff
@@ -0,0 +1,332 @@
+                         yDesc.IsPacked() && meanDesc.IsPacked() && rstdDesc.IsPacked();
+
+    auto dims         = xDesc.GetLengths();
+    size_t grid_size  = 1;
+    size_t outer_size = 1;
+    size_t inner_size = 1;
+    size_t i          = 0;
+    for(; i < normalized_dim; i++)
+    {
+        outer_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    for(; i < dims.size(); i++)
+    {
+        inner_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    auto dtype = xDesc.GetType();
+
+    const std::vector<size_t> vld{LOCAL_SIZE, 1, 1};
+    const std::vector<size_t> vgd{outer_size * vld[0], 1, 1};
+
+    std::string algo_name = "LayerNormForward";
+    std::string network_config =
+        "lnfwd-dtype" + std::to_string(static_cast<int>(dtype)) + "g" + std::to_string(vgd[0]) +
+        "l" + std::to_string(vld[0]) + "normalized_dim" + std::to_string(normalized_dim) + "grid" +
+        std::to_string(grid_size) + "outer_size" + std::to_string(outer_size) + "inner_size" +
+        std::to_string(inner_size) + "xpk" + std::to_string(static_cast<int>(xDesc.IsPacked()));
+    if(weight)
+        network_config += "weightpk" + std::to_string(static_cast<int>(weightDesc.IsPacked()));
+    if(bias)
+        network_config += "biaspk" + std::to_string(static_cast<int>(biasDesc.IsPacked()));
+    network_config += "ypk" + std::to_string(static_cast<int>(yDesc.IsPacked()));
+    if(mean)
+        network_config += "meanpk" + std::to_string(static_cast<int>(meanDesc.IsPacked()));
+    if(rstd)
+        network_config += "rstdpk" + std::to_string(static_cast<int>(rstdDesc.IsPacked()));
+    network_config += "mode" + std::to_string(static_cast<int>(mode)) + "eps" +
+                      std::to_string(static_cast<int>(epsilon));
+
+    std::string program_name = "MIOpenLayerNorm.cpp";
+    std::string kernel_name  = "LayernormFwdContiguous";
+
+    // compile parameters
+    std::string parms =
+        " -DMIOPEN_USE_FP16=" + std::to_string(static_cast<int>(dtype == miopenHalf)) +
+        " -DMIOPEN_USE_FP32=" + std::to_string(static_cast<int>(dtype == miopenFloat)) +
+        " -DMIOPEN_USE_FP64=" + std::to_string(static_cast<int>(dtype == miopenDouble)) +
+        " -DMIOPEN_USE_BF16=" + std::to_string(static_cast<int>(dtype == miopenBFloat16));
+
+    if(mode == MIOPEN_ELEMENTWISE_AFFINE)
+        parms += " -DUSE_MIOPEN_ELEMENTWISE_AFFINE=1";
+    else
+        parms += " -DUSE_MIOPEN_WEIGHT_BIAS=1";
+
+    parms += " -DRUN_FORWARD=1";
+    parms +=
+        " -DIS_INPUT_PACKED=" +
+        std::to_string(static_cast<int>(xDesc.IsPacked() && (!weight || weightDesc.IsPacked()) &&
+                                        (!bias || biasDesc.IsPacked()))) +
+        " -DIS_OUTPUT_PACKED=" +
+        std::to_string(static_cast<int>(yDesc.IsPacked() && (!mean || meanDesc.IsPacked()) &&
+                                        (!rstd || rstdDesc.IsPacked())));
```

Comment:
It seems all that defines are not required, since they are not used in the kernel.
Instead `MIOPEN_TYPE` and `LOCAL_SIZE` should be passed.

Also `-DMIOPEN_USE_FP64=` should be replaced with `-DFSTYPE=` and required type, float or double
```
#if MIOPEN_USE_FP64 == 1
#define FSTYPE double
#else
#define FSTYPE float
#endif
```